### DISCUSSION TITLE
Burst mode rip all track and only do secure rip if AccurateRip check fails

### DISCUSSION THREAD

**Discussion by apastuszak** [whipper-team/whipper#544] [Ideas]
I don't know if Whipper has this feature or not. On my Mac, I am using dBPowerAmp, and it works by doing a burst mode rip of a track and the does an accuraterip comparison,  If it matches, then it knows the rip is good. If it fails, then it will do a secure re-rip of the track. This saved a LOT of time ripping CDs.

**Comment by MerlijnWajer** [comment]
It's definitely possible, I do it in my own private/internal fork of morituri (not whipper, haven't updated it in a while). I'll see if I can get those parts shared here as well (the accurate rip checksum came from that work). I'll have to ask, though.

It's also not too hard to add in I think, if someone feels like it.

**Reply by apastuszak** [reply]
I wish I was a developer, so I could help out with this stuff.

**Comment by singularity098** [comment]
Anybody have any updates on this?  

I've used like half a dozen CD rippers on Linux in the past week and will be going through my entire collection.  None of them were as good as whipper.  It's great!

But as often seems to be the case, it has ONE fatal flaw... it's terribly slow.  I'm taking 20 minutes minimum and up to 60 minutes just to rip a single CD.  I can see that it's making at least 2 passes on every single track.  What is the point of making a second pass if the first pass already matches what's in the AccurateRip database? 

It's not often that a single and relatively simple change to a codebase can offer a 2x performance improvement, with no compromises... and also doesn't go against whipper's first principle of perfect audio duplication.

Anyway... beggers can't be choosers but this is such a good low hanging fruit, I can't believe it's not been done.
